pkg/platform/openstack/loadbalancer.go

```diff
@@ -411,3 +411,35 @@ func listOpenStackOctaviaProviders(client *gophercloud.ServiceClient) ([]provide
 		return providersList, nil
 	}
 }
+
+// Iterate on pool members and check their address against provided list
+// addresses of current master/bootstrap nodes. Remove all surplus members,
+// which address doesn't exists on that list.
+func purgeOpenStackLbPoolMember(client *gophercloud.ServiceClient, poolId string, addresses []string) error {
+	page, err := pools.ListMembers(client, poolId, nil).AllPages()
+	if err != nil {
+		return errors.Wrap(err, "failed to get LB member list")
+	}
+
+	members, err := pools.ExtractMembers(page)
+	if err != nil {
+		return errors.Wrap(err, "failed to extract LB members list")
+	}
+
+	for _, member := range members {
```

Comment:
as we have already added the "new members", does it make sense to compare here members and addresses length first, and if the same assume members are already up to date?

Reply:
Seems tempting, although it won't prevent us from making less Octavia calls, right? And this will ensure, that members addresses are the same as for nodes addresses (as it *should* be) :)

Reply:
I support @gryf here, we could miss kind of a failover of master VM. Regardless of if it's possible or not, it's almost no cost to a do full comparison. It's just golang makes it look long and clumsy. ;)